src/main/java/app/bpartners/geojobs/service/detection/ZoneDetectionJobService.java

```diff
@@ -347,6 +345,13 @@ public ZoneDetectionJob checkHumanDetectionJobStatus(String jobId) {
       humanZDJHealth = UNKNOWN;
     }
 
+    if (FINISHED.equals(humanZDJProgression) && SUCCEEDED.equals(humanZDJHealth)) {
```

Comment:
job and task already have `.isPending()` or `isSucceeded()` or `isFailed()` methods it seems. If not, add these and use its instead.